packages/bsky/src/db/periodic-moderation-action-reversal.ts

```diff
@@ -26,12 +28,34 @@ export class PeriodicModerationActionReversal {
     }
   }
 
-  async revertAction({ id, createdBy }: { id: number; createdBy: string }) {
+  // invert label creation & negations
+  async reverseLabels(labelTxn: LabelService, actionRow: ModerationActionRow) {
+    let uri: string
+    let cid: string | null = null
+
+    if (actionRow.subjectUri && actionRow.subjectCid) {
+      uri = actionRow.subjectUri
+      cid = actionRow.subjectCid
+    } else {
+      uri = actionRow.subjectDid
+    }
+
+    await labelTxn.formatAndCreate(this.appContext.cfg.labelerDid, uri, cid, {
+      create: actionRow.negateLabelVals
+        ? actionRow.negateLabelVals.split(' ')
+        : undefined,
+      negate: actionRow.createLabelVals
+        ? actionRow.createLabelVals.split(' ')
+        : undefined,
+    })
+  }
+
+  async revertAction(actionRow: ModerationActionRow) {
     return this.appContext.db.getPrimary().transaction(async (dbTxn) => {
```

Comment:
While we're in here could we also move the `pushAgent` usage outside of the db transaction?  The issue is relatively small—it just takes-up a database connection that doesn't get used for the duration of `pushAgent.com.atproto.admin.reverseModerationAction()`.